new-packages/ts-project/src/state.ts

```diff
@@ -268,6 +290,34 @@ export function extractState(
                     'description',
                   );
 
+                  const guardV4 = findProperty(ctx, ts, element, 'cond');
+                  const guardV5 = findProperty(ctx, ts, element, 'guard');
+
+                  if (guardV4 && guardV5) {
+                    ctx.errors.push({
+                      type: 'property_mixed',
+                    });
+                  }
+                  const guard = guardV5 ?? guardV4;
+
+                  let guardBlock;
+
+                  if (guard) {
+                    guardBlock = createGuardBlock({
+                      sourceId: ts.isStringLiteralLike(guard.initializer)
+                        ? guard.initializer.text
+                        : `inline:${uniqueId()}`,
+                      parentId: node.uniqueId,
+                    });
+                    ctx.digraph.blocks[guardBlock.uniqueId] = guardBlock;
+                    ctx.digraph.implementations.guards[guardBlock.sourceId] ??=
+                      {
+                        type: 'guard',
+                        id: guardBlock.sourceId,
+                        name: guardBlock.sourceId,
```

Comment:
could you remind me why we need both? 🤔 who they are used by the Studio

Reply:
I'm planning to open a separate PR to tweak block names when inline.